libs/mock-block-dock/src/datastore/use-mock-datastore.ts

```diff
@@ -220,23 +227,62 @@ export const useMockDatastore = (
                   }
                 : undefined;
 
-            const updatedEntity: Entity = {
+            // This is fairly slow, we could leverage something like `immer` to make these partial updates more
```

Comment:
I don't think immer would be faster and would increase bundle size more than a single deep clone function so have suggested tweaking this
```suggestion
            // @todo test performance – if an issue, test against a different deep clone strategy (e.g. [lodash's](https://www.npmjs.com/package/lodash.clonedeep))
```

Reply:
I've updated it, but removed the link and the `todo`. I'd prefer not to add TODOs unless we think they're actually worth doing. If we were trying to investigate slow performance I think we'd find the comment, and I don't think it being a todo makes it much more discoverable, just adds to the endless pit of lost intentions 😅 

[Improve comment](https://github.com/blockprotocol/blockprotocol/pull/931/commits/77e5ceb8ad2bd7d9871543a1d6fe2c74a77c79ac)